math/minuit2/inc/Minuit2/FCNGradAdapter.h

```diff
@@ -78,10 +67,67 @@ class FCNGradAdapter : public FCNGradientBase {
       }
    }
 
+   /// return second derivatives (diagonal of the Hessian matrix)
+   std::vector<double> G2(const std::vector<double> & x) const override {
+      if (fG2Func)
+         return fG2Func(x);
+      if (fHessianFunc) {
+         unsigned int n = fFunc.NDim();
+         if (fG2Vec.empty() ) fG2Vec.resize(n);
+         if (fHessian.empty() ) fHessian.resize(n*n);
+         fHessianFunc(x,fHessian.data());
+         if (!fHessian.empty()) {
+            // get diagonal element of h
+            for (unsigned int i = 0; i < n; i++)
+               fG2Vec[i] = fHessian[i*n+i];
+         }
+         else fG2Vec.clear();
+      }
+      else
+         if (!fG2Vec.empty()) fG2Vec.clear();
+      return fG2Vec;
+   }
+
+   /// compute Hessian. Return Hessian as a std::vector of size(n*n)
+   std::vector<double> Hessian(const std::vector<double> & x ) const override {
+      unsigned int n = fFunc.NDim();
+      if (fHessianFunc) {
+         if (fHessian.empty() ) fHessian.resize(n * n);
+         bool ret = fHessianFunc(x,fHessian.data());
+         if (!ret) {
+            fHessian.clear();
+            std::function<bool(const std::vector<double> &, double *)> emptyFunc;
+            fHessianFunc = emptyFunc;
```

Comment:
```suggestion
            fHessianFunc = nullptr;
```
This is shorter and also works: https://en.cppreference.com/w/cpp/utility/functional/function/operator%3D

Reply:
Thanks , I agree, this is better